annotations/builder/src/main/java/io/sundr/builder/internal/functions/PropertyAs.java

```diff
@@ -213,10 +213,14 @@ public TypeDef apply(Property property) {
       }
       parameters.add(N);
 
+      String prefix = property.hasAttribute(Constants.DESCENDANT_OF)
```

Comment:
I am not sure I understand the logic here.
Why are we adding the diff only when we have an ancestor property ? 
Naming conflicts are not limited there.

Reply:
My understanding is the logic creates synthetic properties for descendents, that's where we want to use the prefixing as the subtypes may collide.  For something that is just a buildable property without multiple subtypes, no prefix is needed - the property name will already be unique.